habitat-lab/habitat/config/default_structured_configs.py

```diff
@@ -186,6 +184,9 @@ class WaypointControlActionConfig(VelocityControlActionConfig):
     max_heading_ang: float = 0.7854  # 45 degrees in rad
     lin_error_tol: float = 0.01
     ang_error_tol: float = 0.025
+    # Stop criteria
+    min_abs_lin_diff: float = 0.01  # meters
```

Comment:
I think we should keep the numbers consistent across `VelocityControlActionConfig` and `WaypointControlActionConfig` , etc. The thresholds for stopping shouldn't change even if you are using a different controller.

Reply:
The reason we cannot have a consistent criterion within the current design is that WaypointControl and VelocityControl are not exactly 1-1 mappings.
VelocityControl executes a base velocity command for 0.1 secs.
WaypointControl runs a controller that generates velocity commands and tries to navigate to the goal for 1 seconds. So that's 10 velocity commands, whether the goal is reached or not. We might get zero velocities but that just means we reached the waypoint prematurely, it does not mean that the intent of the policy is to stay in place (so instead, a stop action is when the waypoint is close to the current location)

Of course, I don't know if this is the best design, so happy to iterate on that